rye/src/lock.rs

```diff
@@ -90,8 +90,11 @@ pub fn update_workspace_lockfile(
     for pyproject_result in workspace.iter_projects() {
         let pyproject = pyproject_result?;
         let rel_url = make_relative_url(&pyproject.root_path(), &workspace.path())?;
-        let applicable_extras = format_project_extras(features_by_project.as_ref(), &pyproject)?;
-        writeln!(local_req_file, "-e {}{}", rel_url, applicable_extras)?;
+        if !workspace.workspace_only() {
+            let applicable_extras =
+                format_project_extras(features_by_project.as_ref(), &pyproject)?;
+            writeln!(local_req_file, "-e {}{}", rel_url, applicable_extras)?;
+        }
```

Comment:
I think this skips installing any workspace member, but it's only the root we want to skip installing (?)

Reply:
Yes indeed, further consideration is needed here.

Reply:
fwiw I think your PR did something useful when I tried it, I inserted the skip root (but keep other members) here and a name in the pyproject file, and then rye sync did what was expected I think - pulling in members and dev-dependencies. Since this is all close to the pip-tools backend (which might be replaced), I think it's tricky to know how to approach this change.